src/main/java/org/jenkinsci/plugins/workflow/graph/StandardGraphLookupView.java

```diff
@@ -221,4 +228,34 @@ public BlockStartNode findEnclosingBlockStart(@Nonnull FlowNode node) {
         }
         return starts;
     }
+
+    @Nonnull
+    @Override
+    public List<FlowNode> getImmediateChildrenForBlockStart(@Nonnull BlockStartNode start) {
```

Comment:
Questions I have here: 

- What about the ordering of children?
- This feels needlessly complicated.  I think there's a need for tree navigation APIs in some sense, but (see downstream PR comments) I don't see it as valid for this one. 

Long-term I kind of lean to storing tree/DAG structural information in the Storage engine itself, with these APIs just delegating to that (that's part of why it is implemented the way it is) -- one reason I don't want to go too much further into this graphLookupView implementation.    It's a placeholder to tide us over for future storage work, ideally.

Reply:
Ah, yeah, the ordering would be inverse here, wouldn't it?

Reply:
Well, back-to-forward in linear flows, but parallels don't have a defined ordering (that's one reason for different scanners).